### PR TITLE
Factor out a (public) function to find a room's predecessor

### PR DESCRIPTION
**Pull request by andybalaam**
This is prep for an implementation of [MSC3946](https://github.com/matrix-org/matrix-spec-proposals/pull/3946)

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: spec/unit/room.spec.ts [hunk 1]
```diff
@@ -3225,4 +3226,52 @@ describe("Room", function () {
...
+                predecessor: { event_id: string; room_id: string } | undefined;
+            } = {
+                "creator": "@daryl:alexandria.example.com",
+                "predecessor": undefined,
+                "m.federate": true,
+                "room_version": "9",
+            };
+            if (predecessorRoomId) {
+                content.predecessor = {
+                    event_id: "spec_is_not_clear_what_id_this_is",
```
Comment (richvdh) [inline spec/unit/room.spec.ts:3245]: For the record: the spec says "The event ID of the last known event in the old room.". Generally it's the tombstone event.

**Review by richvdh — APPROVED** [review]
MSC3946 seems to be under fierce debate, so I'd be hesitant about implementing it right now.

Nevertheless factoring code out to functions is a win!

File: spec/unit/room.spec.ts (hunk 1 above)
Comment (andybalaam) [reply spec/unit/room.spec.ts]: Thank you! Where did you find that? I couldn't see much info in the Client-Server API spec for room upgrades.

Comment (richvdh) [reply spec/unit/room.spec.ts]: on the spec for `m.room.create` (https://spec.matrix.org/v1.5/client-server-api/#mroomcreate)

**Pull request merged**